### PR TITLE
Refactor payments to allow management of methods

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes (partially) #6699.

Note that `Setup required` will be handled in a follow-up and separation of methods is being handled in https://github.com/woocommerce/woocommerce-admin/pull/6766. cc @joelclimbsthings if you need me to make any changes to this in prep for 6766 please let me know.

This PR refactors the payments screen into individual functional components and adds `Set up/Manage/Enable` links.

### Screenshots

Screen Shot 2021-04-12 at 4 36 09 PM

### Detailed test instructions:

1. Do not select "CBD industry" as a store industry during onboarding.
2. Make various payment methods visible by switching to different countries.
3. Attempt to set up various payment methods.
4. Make sure that after setup, a `Manage` link is shown that links to the payment method settings page.
5. Check that simple methods like, cash delivery or bank transfer initially have an `Enable` option.

### CODE REVIEW

File: client/task-list/tasks/payments/recommended-ribbon.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
```
Comment (joelclimbsthings) [inline client/task-list/tasks/payments/recommended-ribbon.js:1]: Just FYI, @elizaan36 has [mentioned the possibility of getting rid of the ribbon](https://github.com/woocommerce/woocommerce-admin/issues/6683#issuecomment-816924780), although we don't need to worry about that here.

File: client/task-list/tasks/payments/index.js [hunk 2]
```diff
@@ -433,27 +108,172 @@ export default compose(
...
+		methods.reduce( ( acc, method ) => {
+			acc[ method.key ] = method.isEnabled;
+			return acc;
+		}, {} )
+	);
+
+	const recommendedMethod = useMemo( () => {
+		const method = methods.find(
+			( m ) =>
+				( m.key === 'wcpay' && m.visible ) ||
```
Comment (joelclimbsthings) [inline client/task-list/tasks/payments/index.js:140]: Looks like these method keys need to be updated I believe?

File: client/task-list/tasks/payments/index.js [hunk 3]
```diff
@@ -433,27 +108,172 @@ export default compose(
...
+					! isConfigured && 'woocommerce-task-payment-not-configured',
+					'woocommerce-task-payment-' + key
+				);
+
+				const isRecommended =
+					key === recommendedMethod && ! isConfigured;
+
+				return (
+					<Card key={ key } className={ classes }>
+						{ isRecommended && key !== 'wcpay' && (
```
Comment (joelclimbsthings) [inline client/task-list/tasks/payments/index.js:234]: Again, need to update this to `woocommerce_payments` I believe.

File: client/task-list/tasks/payments/index.js [hunk 4]
```diff
@@ -433,27 +108,172 @@ export default compose(
...
+							<RecommendedRibbon methodKey={ key } />
+						) }
+						<CardMedia isBorderless>{ before }</CardMedia>
+						<CardBody>
+							<H className="woocommerce-task-payment__title">
+								{ title }
+								{ isRecommended && key === 'wcpay' && (
```
Comment (joelclimbsthings) [inline client/task-list/tasks/payments/index.js:241]: Same here.

**Review by joelclimbsthings — APPROVED** [review]
Hey @joshuatf , good work on this 💯 ! Stoked by the functional component, despite the dicey rebase I just finished. 

I noted a couple spots where the method keys had not been updated, although I've either fixed or wiped out most of these in my soon-to-follow PR, so I'm not too worried about it. Pre-approving so we can get my PR amended as soon as possible as well.

**Comment by joshuatf** [comment]
@joelclimbsthings Added a new `manageUrl` property in https://github.com/woocommerce/woocommerce-admin/pull/6786/commits/1b7f0b643f40c661efd93e1ead279dc1f1ab41c8 since it looks like the keys you mentioned are used in a bunch of places around the app.  It also keeps it more open-ended should a payment method have a non-traditional settings screen in the future.

Also added some changes to e2e tests in https://github.com/woocommerce/woocommerce-admin/pull/6786/commits/0a4fcafe61528081a1b81d99e4d3ec169fbbe5d8 which hopefully passes 🤞

**Comment by joelclimbsthings** [comment]
Great work @joshuatf . I just took everything for another spin, and failed to break anything. Looks/works good! :shipit:

**Pull request merged**